wp-includes/rest-api/auth/class-wp-rest-key-pair.php

```diff
@@ -0,0 +1,798 @@
+<?php
+/**
+ * REST API: WP_REST_Key_Pair class.
+ *
+ * This class is responsible for adding/revoking a users API key:secret pairs.
+ *
+ * This class only allows REST authentication to the `{api_prefix}/wp/v2/token`
+ * endpoint with an API key:secret. This allows a user to be identified via the
+ * REST API without using their login credentials to generate a JSON Web Token.
+ *
+ * @package JWTAuth
+ * @subpackage REST_API
+ * @since 0.1
+ */
+
+/**
+ * Core class used to manage REST API key-pairs which are used to generate JSON Web Tokens.
+ *
+ * @since 0.1
+ */
+class WP_REST_Key_Pair {
+
+	/**
+	 * The user meta key-pair key.
+	 *
+	 * @since 0.1
+	 * @type string
+	 */
+	const _USERMETA_KEY_ = '_key_pairs';
+
+	/**
+	 * The namespace of the authentication route.
+	 *
+	 * @since 0.1
+	 * @type string
+	 */
+	const _NAMESPACE_ = 'wp/v2';
+
+	/**
+	 * The base of the authentication route.
+	 *
+	 * @since 0.1
+	 * @type string
+	 */
+	const _REST_BASE_ = 'key-pair';
+
+	/**
+	 * Initializes the class.
+	 *
+	 * @since 0.1
+	 *
+	 * @see add_action()
+	 * @see add_filter()
+	 */
+	public function init() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ), 99 );
+		add_action( 'show_user_profile', array( $this, 'show_user_profile' ) );
+		add_action( 'edit_user_profile', array( $this, 'show_user_profile' ) );
+		add_action( 'after_password_reset', array( $this, 'after_password_reset' ) );
+		add_action( 'profile_update', array( $this, 'profile_update' ) );
+
+		add_filter( 'rest_authentication_require_token', array( $this, 'require_token' ), 10, 3 );
+		add_filter( 'rest_authentication_user', array( $this, 'authenticate' ), 10, 2 );
+		add_filter( 'rest_authentication_token_private_claims', array( $this, 'payload' ), 10, 2 );
+		add_filter( 'rest_authentication_validate_token', array( $this, 'validate_token' ) );
+	}
+
+	/**
+	 * Return the REST URI for the endpoint.
+	 *
+	 * @since 0.1
+	 *
+	 * @static
+	 */
+	public static function get_rest_uri() {
+		return sprintf( '/%s/%s/%s', rest_get_url_prefix(), self::_NAMESPACE_, self::_REST_BASE_ );
+	}
+
+	/**
+	 * Registers the routes for the authentication method.
+	 *
+	 * @since 0.1
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		$args = array(
+			array(
+				'methods'  => WP_REST_Server::CREATABLE,
+				'callback' => array(
```

Comment:
Suggestion: leave in one line: `'callback' => array( $this, 'generate_key_pair' )` This and others below.